api/src/main/java/org/pmiops/workbench/cohortbuilder/util/BQParameterUtil.java

```diff
@@ -1,16 +1,78 @@
 package org.pmiops.workbench.cohortbuilder.util;
 
+import com.google.cloud.bigquery.QueryJobConfiguration;
 import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import java.time.Instant;
+import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 public final class BQParameterUtil {
+  private static final int MICROSECONDS_IN_MILLISECOND = 1000;
 
   /** Generate a unique parameter name and add it to the parameter map provided. */
   public static String buildParameter(
       Map<String, QueryParameterValue> queryParameterValueMap,
       QueryParameterValue queryParameterValue) {
     String parameterName = "p" + queryParameterValueMap.size();
     queryParameterValueMap.put(parameterName, queryParameterValue);
+    return decorateParameterName(parameterName);
+  }
+
+  public static QueryParameterValue instantToQPValue(Instant instant) {
+    return QueryParameterValue.timestamp(instant.toEpochMilli() * MICROSECONDS_IN_MILLISECOND);
+  }
+
+  // Since BigQuery doesn't expose the literal query string built from a QueryJobConfiguration,
+  // this method does the next best thing. Useful for diagnostics, logging, testing, etc.
+  public static String getReplacedQueryText(QueryJobConfiguration queryJobConfiguration) {
+    String result = "-- reconstructed query text\n" + queryJobConfiguration.getQuery();
+    final Map<String, QueryParameterValue> keyToNamedParameter =
+        queryJobConfiguration.getNamedParameters().entrySet().stream()
+            .collect(Collectors.toMap(e -> decorateParameterName(e.getKey()), Entry::getValue));
+
+    // Sort in reverse lenght order so we don't partially replace any parameter names (e.g. replace
+    // "@foo" before "@foo_bar").
+    final List<String> keysByLengthDesc =
```

Comment:
It just now occurred to me that I can use regex Patterns with word boundaries to solve this issue, and avoid re-sorting the whole list and lots of this code. All in favor of less code in this function?